### PR TITLE
DOC: update the pandas.Index.drop_duplicates and pandas.Series.drop_duplicates docstring

### PR DESCRIPTION
**Pull request by DaanVanHauwermeiren**
Checklist for the pandas documentation sprint (ignore this if you are doing
an unrelated PR):

- [X] PR title is "DOC: update the <your-function-or-method> docstring"
- [ ] The validation script passes: `scripts/validate_docstrings.py <your-function-or-method>`
- [X] The PEP8 style check passes: `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [X] The html version looks good: `python doc/make.py --single <your-function-or-method>`
- [ ] It has been proofread on language by another sprint participant

Please include the output of the validation script below between the "```" ticks:

Method ```pandas.Index.drop_duplicates```:

```
################################################################################
################### Docstring (pandas.Index.drop_duplicates) ###################
################################################################################

Return Index with duplicate values removed.

The drop_duplicates method can remove occurences or whole sets
of duplicated entries in a pandas.Index object.

Parameters
----------
keep : {'first', 'last', ``False``}, default 'first'
    - 'first' : Drop duplicates except for the first occurrence.
    - 'last' : Drop duplicates except for the last occurrence.
    - ``False`` : Drop all duplicates.

Returns
-------
deduplicated : Index

See Also
--------
pandas.Series.drop_duplicates : equivalent method on pandas.Series

Examples
--------
Generate an pandas.Index with duplicate values.

>>> idx = pd.Index(['lama', 'cow', 'lama', 'beetle', 'lama', 'hippo'])

With the 'keep' parameter, the selection behaviour of duplicated values
can be changed. The value 'first' keeps the first occurrence for each
set of duplicated entries. The default value of keep is 'first'.

>>> idx.drop_duplicates(keep='first')
Index(['lama', 'cow', 'beetle', 'hippo'], dtype='object')

The value 'last' keeps the last occurrence for each set of duplicated
entries.

>>> idx.drop_duplicates(keep='last')
Index(['cow', 'beetle', 'lama', 'hippo'], dtype='object')

The value ``False`` discards all sets of duplicated entries.

>>> idx.drop_duplicates(keep=False)
Index(['cow', 'beetle', 'hippo'], dtype='object')

################################################################################
################################## Validation ##################################
################################################################################

Errors found:
	Errors in parameters section
		Parameter "keep" description should start with capital letter

```

Method ```pandas.Series.drop_duplicates```:

```

################################################################################
################## Docstring (pandas.Series.drop_duplicates)  ##################
################################################################################

Return Series with duplicate values removed.

The drop_duplicates method can remove occurences or whole sets
of duplicated entries in a pandas.Series object.

Parameters
----------
keep : {'first', 'last', ``False``}, default 'first'
    - 'first' : Drop duplicates except for the first occurrence.
    - 'last' : Drop duplicates except for the last occurrence.
    - ``False`` : Drop all duplicates.
inplace : boolean, default ``False``
    If ``True``, performs operation inplace and returns None.

Returns
-------
deduplicated : Series

See Also
--------
pandas.Index.drop_duplicates : equivalent method on pandas.Index

Examples
--------
Generate an Series with duplicated entries.

>>> s = pd.Series(['lama', 'cow', 'lama', 'beetle', 'lama', 'hippo'],
...               name='animal')
>>> s
0      lama
1       cow
2      lama
3    beetle
4      lama
5     hippo
Name: animal, dtype: object

With the 'keep' parameter, the selection behaviour of duplicated values
can be changed. The value 'first' keeps the first occurrence for each
set of duplicated entries. The default value of keep is 'first'.

>>> s.drop_duplicates()
0      lama
1       cow
3    beetle
5     hippo
Name: animal, dtype: object

The value 'last' for parameter 'keep' keeps the last occurrence for
each set of duplicated entries.

>>> s.drop_duplicates(keep='last')
1       cow
3    beetle
4      lama
5     hippo
Name: animal, dtype: object

The value ``False`` for parameter 'keep' discards all sets of
duplicated entries. Setting the value of 'inplace' to ``True`` performs
the operation inplace and returns ``None``.

>>> s.drop_duplicates(keep=False, inplace=True)
>>> s
1       cow
3    beetle
5     hippo
Name: animal, dtype: object

################################################################################
################################## Validation ##################################
################################################################################

Errors found:
	Errors in parameters section
		Parameter "keep" description should start with capital letter


```

If the validation script still gives errors, but you think there is a good reason
to deviate in this case (and there are certainly such cases), please state this
explicitly.

The parameter keyword fails in the validation script. In the DOC sprint team, it was decided that summing up the parameter values is more valuable.

### CODE REVIEW

File: pandas/core/indexes/base.py [hunk 1]
```diff
@@ -4017,8 +4017,52 @@ def unique(self, level=None):
...
+            - 'last' : Drop duplicates except for the last occurrence.
+            - ``False`` : Drop all duplicates.
+
+        Returns
+        -------
+        deduplicated : Index
+
+        See Also
+        --------
+        pandas.Series.drop_duplicates : equivalent method on pandas.Series
```
Comment (TomAugspurger) [inline pandas/core/indexes/base.py]: I think just `Series.drop_duplicates`

File: pandas/core/indexes/base.py [hunk 2]
```diff
@@ -4017,8 +4017,52 @@ def unique(self, level=None):
         result = super(Index, self).unique()
         return self._shallow_copy(result)
 
-    @Appender(base._shared_docs['drop_duplicates'] % _index_doc_kwargs)
     def drop_duplicates(self, keep='first'):
+        """
+        Return Index with duplicate values removed.
+
+        The drop_duplicates method can remove occurences or whole sets
```
Comment (TomAugspurger) [inline pandas/core/indexes/base.py]: I'm not sure what "whole sets" means here. Personally I think this extended summary can be removed. The first line is descriptive enough.

File: pandas/core/indexes/base.py [hunk 3]
```diff
@@ -4017,8 +4017,52 @@ def unique(self, level=None):
...
+        --------
+        pandas.Series.drop_duplicates : equivalent method on pandas.Series
+
+        Examples
+        --------
+        Generate an pandas.Index with duplicate values.
+
+        >>> idx = pd.Index(['lama', 'cow', 'lama', 'beetle', 'lama', 'hippo'])
+
+        With the 'keep' parameter, the selection behaviour of duplicated values
```
Comment (TomAugspurger) [inline pandas/core/indexes/base.py]: Change first sentence to

    The `keep` parameter controls  which duplicate values are removed.

**Review by TomAugspurger — COMMENTED** [review]
ping @jorisvandenbossche do we want to split this shared doc? Or keep the shared doc and just have an `%(examples)s` section?

**Comment by jorisvandenbossche** [comment]
Well, we were having this discussion here as well :-) 
I said them to start with making both, and then we could afterwards see what they would really have in common. In this case it is mainly the Parameters section. So we could also have a variables with this text, and inject only that part in both docstrings. Not sure if that will be simpler though

File: pandas/core/indexes/base.py (hunk 2 above)
Comment (DaanVanHauwermeiren) [reply pandas/core/indexes/base.py]: Ok, thanks Tom, I'll adjust accordingly!

File: pandas/core/indexes/base.py (hunk 1 above)
Comment (DaanVanHauwermeiren) [reply pandas/core/indexes/base.py]: updated in d92c124dd5668bb7d983473466fb3355f8dd5f5c

File: pandas/core/indexes/base.py (hunk 3 above)
Comment (DaanVanHauwermeiren) [reply pandas/core/indexes/base.py]: sentence updated in 0d604ac

File: pandas/core/series.py [hunk 4]
```diff
@@ -1316,8 +1316,74 @@ def unique(self):
...
+            If ``True``, performs operation inplace and returns None.
+
+        Returns
+        -------
+        deduplicated : Series
+
+        See Also
+        --------
+        Index.drop_duplicates : equivalent method on Index
+
```
Comment (jreback) [inline pandas/core/series.py]: can also DataFrame.drop_duplicates. and Series.duplicated

Comment (jreback) [reply pandas/core/series.py]: can you edit DataFrame.drop_duplicates if needed to ensure the back-link to here (unless other PR)

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/series.py (hunk 4 above)
Comment (jorisvandenbossche) [reply pandas/core/series.py]: let's keep that for another PR. But good idea to also add DataFrame.drop_duplicates and Series.duplicated

Comment (DaanVanHauwermeiren) [reply pandas/core/series.py]: adjusted in c300ea6

**Review by jreback — APPROVED** [review]

**Review by jorisvandenbossche — APPROVED** [review]

File: pandas/core/indexes/base.py [hunk 5]
```diff
@@ -4017,8 +4017,49 @@ def unique(self, level=None):
...
+            - 'last' : Drop duplicates except for the last occurrence.
+            - ``False`` : Drop all duplicates.
+
+        Returns
+        -------
+        deduplicated : Index
+
+        See Also
+        --------
+        Series.drop_duplicates : equivalent method on Series
```
Comment (jorisvandenbossche) [inline pandas/core/indexes/base.py]: can you update see also here as well?

Comment (DaanVanHauwermeiren) [reply pandas/core/indexes/base.py]: adjusted in 8763f33

**Pull request merged**